bundles/org.jkiss.utils/tests/org/jkiss/utils/BeanUtilsTest.java

```diff
@@ -0,0 +1,218 @@
+        Assert.assertNull(
+                BeanUtils.getGetMethod(String.class, "length", true));
+        Assert.assertNull(
+                BeanUtils.getGetMethod(String.class, "length", false));
+    }
+
+    @Test
+    public void testMethodNameToPropertyName() {
+        Assert.assertNull(BeanUtils.methodNameToPropertyName(""));
+        Assert.assertEquals("g", BeanUtils.methodNameToPropertyName("G"));
+        Assert.assertEquals("get", BeanUtils.methodNameToPropertyName("Get"));
+        Assert.assertEquals("empty",
+                BeanUtils.methodNameToPropertyName("isEmpty"));
+        Assert.assertEquals("length",
+                BeanUtils.methodNameToPropertyName("getlength"));
+        Assert.assertEquals("length",
+                BeanUtils.methodNameToPropertyName("setlength"));
+    }
+
+    @Test
+    public void testIsArrayType() {
+        Assert.assertFalse(BeanUtils.isArrayType(String.class));
+    }
+
+    @Test
+    public void testIsCollectionType() {
+        Assert.assertFalse(BeanUtils.isCollectionType(String.class));
+    }
+
+    @Test
+    public void testGetCollectionType1() {
+        ParameterizedType parameterizedType =
+                PowerMockito.mock(ParameterizedType.class);
+        PowerMockito.when(parameterizedType.getActualTypeArguments())
+                .thenReturn(new Type[]{String.class});
+
+        Assert.assertEquals(String.class,
+                BeanUtils.getCollectionType(parameterizedType));
+
+        Assert.assertNull(BeanUtils.getCollectionType(null));
+    }
+
+    @Test
+    public void testGetCollectionType2() {
+        WildcardType wildcardType = PowerMockito.mock(WildcardType.class);
+        PowerMockito.when(wildcardType.getUpperBounds())
+                .thenReturn(new Type[]{String.class, Integer.class});
+
+        ParameterizedType parameterizedType =
+                PowerMockito.mock(ParameterizedType.class);
+        PowerMockito.when(parameterizedType.getActualTypeArguments())
+                .thenReturn(new Type[]{wildcardType});
+
+        Assert.assertEquals(String.class,
+                BeanUtils.getCollectionType(parameterizedType));
+    }
+
+    @Test
+    public void testGetCollectionType3() {
+        WildcardType wildcardType = PowerMockito.mock(WildcardType.class);
+        PowerMockito.when(wildcardType.getUpperBounds())
+                .thenReturn(new Type[0]);
+        PowerMockito.when(wildcardType.getLowerBounds())
+                .thenReturn(new Type[]{String.class, Integer.class});
+
+        ParameterizedType parameterizedType =
+                PowerMockito.mock(ParameterizedType.class);
+        PowerMockito.when(parameterizedType.getActualTypeArguments())
+                .thenReturn(new Type[]{wildcardType});
+
+        Assert.assertEquals(String.class,
+                BeanUtils.getCollectionType(parameterizedType));
+    }
+
+    @Test
+    public void testReadObjectProperty()
```

Comment:
Can you assert any non-null values here?

Reply:
Done